R/data.table.R

```diff
@@ -128,6 +128,35 @@ replace_dot_alias = function(e) {
   }
 }
 
+.reassign_extracted_table = function(name, value, env = parent.frame(), err_msg_na, err_msg_detail = NULL) {
+  k = eval(name[[2L]], env, env)
+  if (is.list(k)) {
+    origj = j = if (name %iscall% "$") as.character(name[[3L]]) else eval(name[[3L]], env, env)
+    if (length(j) != 1L) {
```

Comment:
We could eliminate this cruft by just supporting such case, I don't imagine it would be that hard? (famous last words)